core/templates/pages/delete-account-page/services/delete-account-backend-api.service.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend Api Service for Delete Account Page.
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DeleteAccountBackendApiService {
+  constructor(
+    private siteAnalyticsService: SiteAnalyticsService,
+    private windowRef: WindowRef,
+    private http: HttpClient,
+  ) {}
+
+  deleteAccount(): void {
+    // eslint-disable-next-line dot-notation
```

Comment:
Why is this added? Can we do it without this?

Reply:
Hmm but for delete, I have seen the usage of `['delete']` too, could you please unify it across the codebase (it should be a few files and if you decide to go with the dot notation then probably always add an explanation and not just the disable comment.

Btw, it is fine to do this in a separate PR if this needs to be merged quickly but if you decide to do that please create an issue about it.

cc @srijanreddy98

Reply:
made changes here since there were a few only,PTAL

Reply:
Sorry, thinking about this do we know why eslint reports it as an error? Would using `@typescript-eslint/dot-notation` help? Or is there any other way how to fix this in more general way?

Reply:
We need to set allowedKeywords to true as by default it requires bracket notation for all property access if the property happens to be a keyword, like delete. I've left a comment above to fix this issue!